### DISCUSSION TITLE
ScriptReference/Vector3

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#37] [Notes]
# ScriptReference/Vector3

https://docs.unity3d.com/ScriptReference/Vector3.html

**Comment by vertxxyz** [comment]
When debugging vectors it's important to note that the displayed fractional precision is extremely low due to the default implementation of [`ToString`](https://github.com/Unity-Technologies/UnityCsReference/blob/master/Runtime/Export/Math/Vector3.cs#L497). This rounding can make vectors look identical when in reality they are relatively far apart.
You can instead use `vector.ToString("FX")` where `X` is the amount of fractional precision you want.

**Comment by adammyhre** [comment]
Missing from the documentation is the non-static version of the public method `Normalize`.  This method makes this vector have a magnitude of 1, mutating the instance.
- `public static Vector3 Normalize(Vector3 value)` Makes a vector have a magnitude of 1.
- `public void Normalize()` Makes THIS vector have a magnitude of 1.
